client/src/App.jsx

```diff
@@ -64,27 +84,32 @@ class App extends React.Component {
     }
 
     history.listen();
+    var windowLocationSearch = window.location.search;
+    var windowLocationHash = window.location.hash;
+
+    var urlStateInfo;
 
     $.ajax({
       method: 'GET',
       url: `/${productId}/availableDates`,
       success: (dates) => {
-        this.setState({
-          dates: dates
-        });
```

Comment:
Perhaps consider naming these ajax functions and moving the definition outside of componentDidMount(). For example, fetchAvailableDates() or something?

That could make it easier to digest at a glance what is happening during the componentDidMount() lifecycle method, and then someone could dig into the function details only if needed.